OIDN configure fails due to C++17 requirement

### Describe the bug

After the latest [pulldown](https://github.com/intel/llvm/pull/22020), we now require C++17.
This causes OIDN configure to [fail](https://github.com/intel/llvm/actions/runs/26076141895/job/76758885261):
```
CMake Error at devices/sycl/CMakeLists.txt:25 (message):
  Building with SYCL support requires oneAPI DPC++ Compiler as the C/C++
  compiler
```

From `--debug-trycompile`
```
[1/2] Building CXX object CMakeFiles/cmTC_9ec30.dir/src.cxx.o
FAILED: CMakeFiles/cmTC_9ec30.dir/src.cxx.o 
ccache /iusers/nsarnie/llvm_oidn/build/bin/clang++ -DOIDN_DEVICE_SYCL_SUPPORTED  -Wall -Wno-unknown-pragmas -Wno-pass-failed -Wno-unneeded-internal-declaration -fPIC -Wformat -Wformat-security     -fsycl -std=c++11 -MD -MT CMakeFiles/cmTC_9ec30.dir/src.cxx.o -MF CMakeFiles/cmTC_9ec30.dir/src.cxx.o.d -o CMakeFiles/cmTC_9ec30.dir/src.cxx.o -c /iusers/nsarnie/oidn/build/CMakeFiles/CMakeScratch/TryCompile-d47die/src.cxx
clang++: error: SYCL requires C++17 or later; '-std=c++11' is not supported
clang++: error: SYCL requires C++17 or later; '-std=c++11' is not supported
ninja: build stopped: subcommand failed.
```

My understanding after talking with the Driver team is this was supposed to be an existing requirement, but there was a bug in intel/llvm allowing older C++ versions and it was only caught when we implemented the requirement upstream.

### To reproduce

_No response_

### Environment

_No response_

### Additional context

_No response_

--- a/devops/actions/blender/oidn/action.yml
+++ b/devops/actions/blender/oidn/action.yml
@@ -28,7 +28,7 @@ runs:
       uses: ./devops/actions/cached_checkout
       with:
         repository: 'RenderKit/oidn'
-        ref: 'fbd9370fe3f863400dc201ce7115d2e7be0d0cb4' # Switch to release branch after next release, see https://github.com/intel/llvm/issues/21893
+        ref: '4e92767dd4a46574a968faa1a3d6c62af699ee38' # Switch to release branch after next release, see https://github.com/intel/llvm/issues/21893
         path: 'oidn-src'
         cache_path: ${{ runner.os == 'Windows' && 'D:\\github\\_work\\repo_cache\\' || '/__w/repo_cache/' }}
 

